api/token.go

```diff
@@ -562,7 +571,7 @@ func (a *API) IdTokenGrant(ctx context.Context, w http.ResponseWriter, r *http.R
 	return sendJSON(w, http.StatusOK, token)
 }
 
-func generateAccessToken(tx *storage.Connection, user *models.User, sessionId *uuid.UUID, expiresIn time.Duration, secret string) (string, error) {
+func generateAccessToken(tx *storage.Connection, user *models.User, sessionId *uuid.UUID, providerId *string, expiresIn time.Duration, secret string) (string, error) {
```

Comment:
At some point perhaps we can consider refactoring all the params into a single struct